app-ng/src/job.lisp

```diff
@@ -0,0 +1,145 @@
+(in-package #:qvm-app-ng)
+
+(adt:defdata job-status
+  job-status-fresh
+  job-status-running
+  job-status-killed
+  (job-status-finished t)
+  (job-status-error condition))
+
+(defun job-status-name (job)
+  (adt:match job-status (job-status job)
+    (job-status-fresh "fresh")
+    (job-status-running "running")
+    (job-status-killed "killed")
+    ((job-status-finished _) "finished")
+    ((job-status-error _) "error")))
+
+(defstruct (job (:constructor %make-job (work-function)))
```

Comment:
style-warning here about not being able to inline the call to `job-status` in `job-status-name`, above. Move this `defstruct` above `job-status-name` to avoid.